<?xml version="1.0"?>

<sdf version='1.9'>
  <world name='empty'>
    <physics name='1ms' type='ignored'>
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    <plugin name='gz::sim::systems::Physics' filename='ignition-gazebo-physics-system'/>
    <plugin name='gz::sim::systems::UserCommands' filename='ignition-gazebo-user-commands-system'/>
    <plugin name='gz::sim::systems::SceneBroadcaster' filename='ignition-gazebo-scene-broadcaster-system'/>
    <plugin name='gz::sim::systems::Contact' filename='ignition-gazebo-contact-system'/>
    <plugin name='gz::sim::systems::Sensors' filename='ignition-gazebo6-sensors-system'/>
    <plugin name='gz::sim::systems::UserCommands' filename='ignition-gazebo-user-commands-system'/>
    <gravity>0 0 -9.8</gravity>
    <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
    <atmosphere type='adiabatic'/>
    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>true</shadows>
    </scene>
    <model name='marble_husky_sensor_config_5'>
      <plugin name='ignition::gazebo::systems::DiffDrive' filename='libignition-gazebo-diff-drive-system.so'>
        <left_joint>front_left_wheel_joint</left_joint>
        <left_joint>rear_left_wheel_joint</left_joint>
        <right_joint>front_right_wheel_joint</right_joint>
        <right_joint>rear_right_wheel_joint</right_joint>
        <wheel_separation>0.535</wheel_separation>
        <wheel_radius>0.165</wheel_radius>
        <odom_publish_frequency>1</odom_publish_frequency>
        <topic>cmd_vel</topic>
      </plugin>
      <link name='base_link'>
        <inertial>
          <pose>0 0 0.13228 0 -0 0</pose>
          <mass>50</mass>
          <inertia>
            <ixx>0.61539699999999997</ixx>
            <ixy>-0.0240585</ixy>
            <ixz>-0.120749</ixz>
            <iyy>1.7538800000000001</iyy>
            <iyz>-0.0028322</iyz>
            <izz>2.0364100000000001</izz>
          </inertia>
        </inertial>
        <collision name='base_link_collision'>
          <pose>0 0 0.12 0 0 0</pose>
          <geometry>
            <box>
              <size>0.9874 0.5709 0.05</size>
            </box>
          </geometry>
          <surface>
            <friction>
              <ode/>
            </friction>
            <bounce/>
            <contact/>
          </surface>
        </collision>
        <collision name='base_link_collision_bottom'>
          <pose>0 0 0.046 0 0 0</pose>
          <geometry>
            <box>
              <size>0.8 0.5709 0.095</size>
            </box>
          </geometry>
          <surface>
            <friction>
              <ode/>
            </friction>
            <bounce/>
            <contact/>
          </surface>
        </collision>
        <collision name='base_link_collision_1'>
          <pose>0 0 0.185625 0 -0 0</pose>
          <geometry>
            <box>
              <size>0.78992 0.5709 0.12375</size>
            </box>
          </geometry>
          <surface>
            <friction>
              <ode/>
            </friction>
            <bounce/>
            <contact/>
          </surface>
        </collision>
        <visual name='base_link_visual'>
          <geometry>
            <mesh>
              <scale>1 1 1</scale>
              <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/MARBLE_HUSKY_SENSOR_CONFIG_4/tip/files/meshes/base_link.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <visual name='base_link_fixed_joint_lump__front_bumper_link_visual_3'>
          <pose>0.48 0 0.091 0 -0 0</pose>
          <geometry>
            <mesh>
              <scale>1 1 1</scale>
              <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/MARBLE_HUSKY_SENSOR_CONFIG_4/tip/files/meshes/bumper.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <visual name='base_link_fixed_joint_lump__rear_bumper_link_visual_4'>
          <pose>-0.48 0 0.091 0 -0 3.14159</pose>
          <geometry>
            <mesh>
              <scale>1 1 1</scale>
              <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/MARBLE_HUSKY_SENSOR_CONFIG_4/tip/files/meshes/bumper.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <visual name='base_link_fixed_joint_lump__top_chassis_link_visual_5'>
          <geometry>
            <mesh>
              <scale>1 1 1</scale>
              <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/MARBLE_HUSKY_SENSOR_CONFIG_4/tip/files/meshes/top_chassis.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <visual name='base_link_fixed_joint_lump__user_rail_link_visual_13'>
          <pose>0.272 0 0.245 0 -0 0</pose>
          <geometry>
            <mesh>
              <scale>1 1 1</scale>
              <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/MARBLE_HUSKY_SENSOR_CONFIG_4/tip/files/meshes/user_rail.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <gravity>true</gravity>
        <velocity_decay/>
        <sensor name='imu_sensor' type='imu'>
          <pose>0 0 0 0 -0 0</pose>
          <topic>__default__</topic>
          <update_rate>200</update_rate>
          <enable_metrics>false</enable_metrics>
          <imu>
            <orientation_reference_frame>
              <localization>CUSTOM</localization>
              <custom_rpy>0 0 0</custom_rpy>
              <grav_dir_x>1 0 0</grav_dir_x>
            </orientation_reference_frame>
            <angular_velocity>
              <x>
                <noise type='gaussian'>
                  <mean>0</mean>
                  <stddev>0.0089999999999999993</stddev>
                  <bias_mean>0.00075000000000000002</bias_mean>
                  <bias_stddev>0.0050000000000000001</bias_stddev>
                  <dynamic_bias_stddev>2.0000000000000002e-05</dynamic_bias_stddev>
                  <dynamic_bias_correlation_time>400</dynamic_bias_correlation_time>
                  <precision>0.00025000000000000001</precision>
                </noise>
              </x>
              <y>
                <noise type='gaussian'>
                  <mean>0</mean>
                  <stddev>0.0089999999999999993</stddev>
                  <bias_mean>0.00075000000000000002</bias_mean>
                  <bias_stddev>0.0050000000000000001</bias_stddev>
                  <dynamic_bias_stddev>2.0000000000000002e-05</dynamic_bias_stddev>
                  <dynamic_bias_correlation_time>400</dynamic_bias_correlation_time>
                  <precision>0.00025000000000000001</precision>
                </noise>
              </y>
              <z>
                <noise type='gaussian'>
                  <mean>0</mean>
                  <stddev>0.0089999999999999993</stddev>
                  <bias_mean>0.00075000000000000002</bias_mean>
                  <bias_stddev>0.0050000000000000001</bias_stddev>
                  <dynamic_bias_stddev>2.0000000000000002e-05</dynamic_bias_stddev>
                  <dynamic_bias_correlation_time>400</dynamic_bias_correlation_time>
                  <precision>0.00025000000000000001</precision>
                </noise>
              </z>
            </angular_velocity>
            <linear_acceleration>
              <x>
                <noise type='gaussian'>
                  <mean>0</mean>
                  <stddev>0.021000000000000001</stddev>
                  <bias_mean>0.050000000000000003</bias_mean>
                  <bias_stddev>0.0074999999999999997</bias_stddev>
                  <dynamic_bias_stddev>0.00037500000000000001</dynamic_bias_stddev>
                  <dynamic_bias_correlation_time>175</dynamic_bias_correlation_time>
                  <precision>0.0050000000000000001</precision>
                </noise>
              </x>
              <y>
                <noise type='gaussian'>
                  <mean>0</mean>
                  <stddev>0.021000000000000001</stddev>
                  <bias_mean>0.050000000000000003</bias_mean>
                  <bias_stddev>0.0074999999999999997</bias_stddev>
                  <dynamic_bias_stddev>0.00037500000000000001</dynamic_bias_stddev>
                  <dynamic_bias_correlation_time>175</dynamic_bias_correlation_time>
                  <precision>0.0050000000000000001</precision>
                </noise>
              </y>
              <z>
                <noise type='gaussian'>
                  <mean>0</mean>
                  <stddev>0.021000000000000001</stddev>
                  <bias_mean>0.050000000000000003</bias_mean>
                  <bias_stddev>0.0074999999999999997</bias_stddev>
                  <dynamic_bias_stddev>0.00037500000000000001</dynamic_bias_stddev>
                  <dynamic_bias_correlation_time>175</dynamic_bias_correlation_time>
                  <precision>0.0050000000000000001</precision>
                </noise>
              </z>
            </linear_acceleration>
            <enable_orientation>true</enable_orientation>
          </imu>
        </sensor>
        <light name='flashlight_flashlight_light_source_lamp_light_front' type='spot'>
          <pose>0.5179 0 0.53718 -3e-06 1.57069 -3.14159</pose>
          <cast_shadows>false</cast_shadows>
          <intensity>1</intensity>
          <direction>0 0 -1</direction>
          <diffuse>0.8 0.8 0.5 1</diffuse>
          <specular>0.8 0.8 0.5 1</specular>
          <attenuation>
            <range>35</range>
            <linear>0</linear>
            <constant>0.10000000000000001</constant>
            <quadratic>0.0025000000000000001</quadratic>
          </attenuation>
          <spot>
            <inner_angle>1.8</inner_angle>
            <outer_angle>1.8999999999999999</outer_angle>
            <falloff>1</falloff>
          </spot>
        </light>
        <light name='flashlight_flashlight_light_source_lamp_light_left' type='spot'>
          <pose>0.4379 0.12 0.53718 1.5708 -0 0</pose>
          <cast_shadows>false</cast_shadows>
          <intensity>1</intensity>
          <direction>0 0 -1</direction>
          <diffuse>0.8 0.8 0.5 1</diffuse>
          <specular>0.8 0.8 0.5 1</specular>
          <attenuation>
            <range>35</range>
            <linear>0</linear>
            <constant>0.10000000000000001</constant>
            <quadratic>0.0025000000000000001</quadratic>
          </attenuation>
          <spot>
            <inner_angle>1.8</inner_angle>
            <outer_angle>1.8999999999999999</outer_angle>
            <falloff>1</falloff>
          </spot>
        </light>
        <light name='flashlight_flashlight_light_source_lamp_light_right' type='spot'>
          <pose>0.4379 -0.12 0.53718 -1.5708 0 0</pose>
          <cast_shadows>false</cast_shadows>
          <intensity>1</intensity>
          <direction>0 0 -1</direction>
          <diffuse>0.8 0.8 0.5 1</diffuse>
          <specular>0.8 0.8 0.5 1</specular>
          <attenuation>
            <range>35</range>
            <linear>0</linear>
            <constant>0.10000000000000001</constant>
            <quadratic>0.0025000000000000001</quadratic>
          </attenuation>
          <spot>
            <inner_angle>1.8</inner_angle>
            <outer_angle>1.8999999999999999</outer_angle>
            <falloff>1</falloff>
          </spot>
        </light>
        <light name='flashlight_flashlight_light_source_lamp_light_back' type='spot'>
          <pose>0.3579 0 0.53718 -3e-06 -1.57069 3.14159</pose>
          <cast_shadows>false</cast_shadows>
          <intensity>1</intensity>
          <direction>0 0 -1</direction>
          <diffuse>0.8 0.8 0.5 1</diffuse>
          <specular>0.8 0.8 0.5 1</specular>
          <attenuation>
            <range>35</range>
            <linear>0</linear>
            <constant>0.10000000000000001</constant>
            <quadratic>0.0025000000000000001</quadratic>
          </attenuation>
          <spot>
            <inner_angle>1.8</inner_angle>
            <outer_angle>1.8999999999999999</outer_angle>
            <falloff>1</falloff>
          </spot>
        </light>
        <pose>0 0 0 0 -0 0</pose>
        <enable_wind>false</enable_wind>
      </link>
      <link name='front_left_wheel_link'>
        <pose>0.256 0.2854 0.03282 -0 -1.07543 0</pose>
        <inertial>
          <mass>2.637</mass>
          <inertia>
            <ixx>0.024670000000000001</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.044110000000000003</iyy>
            <iyz>0</iyz>
            <izz>0.024670000000000001</izz>
          </inertia>
          <pose>0 0 0 0 -0 0</pose>
        </inertial>
        <collision name='front_left_wheel_link_collision'>
          <pose>0 0 0 1.5707963267948966 -0 0</pose>
          <geometry>
            <cylinder>
              <length>0.1143</length>
              <radius>0.1651</radius>
            </cylinder>
          </geometry>
          <surface>
            <contact>
              <ode>
                <kp>10000000</kp>
                <kd>1</kd>
              </ode>
            </contact>
            <friction>
              <ode>
                <mu>0.5</mu>
                <mu2>1</mu2>
                <slip1>0.00062</slip1>
                <slip2>0</slip2>
                <fdir1>0 0 1</fdir1>
              </ode>
            </friction>
            <bounce/>
          </surface>
        </collision>
        <visual name='front_left_wheel_link_visual'>
          <geometry>
            <mesh>
              <scale>1 1 1</scale>
              <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/MARBLE_HUSKY_SENSOR_CONFIG_4/tip/files/meshes/wheel.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <gravity>true</gravity>
        <velocity_decay/>
        <enable_wind>false</enable_wind>
      </link>
      <joint name='front_left_wheel_joint' type='revolute'>
        <child>front_left_wheel_link</child>
        <parent>base_link</parent>
        <axis>
          <xyz>0 1 0</xyz>
          <limit>
            <lower>-10000000000000000</lower>
            <upper>10000000000000000</upper>
            <effort>inf</effort>
            <velocity>inf</velocity>
            <stiffness>100000000</stiffness>
            <dissipation>1</dissipation>
          </limit>
          <dynamics>
            <spring_reference>0</spring_reference>
            <spring_stiffness>0</spring_stiffness>
            <damping>0</damping>
            <friction>0</friction>
          </dynamics>
        </axis>
        <pose>0 0 0 0 -0 0</pose>
      </joint>
      <link name='front_right_wheel_link'>
        <pose>0.256 -0.2854 0.03282 -3.14159 0.919859 -3.14159</pose>
        <inertial>
          <mass>2.637</mass>
          <inertia>
            <ixx>0.024670000000000001</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.044110000000000003</iyy>
            <iyz>0</iyz>
            <izz>0.024670000000000001</izz>
          </inertia>
          <pose>0 0 0 0 -0 0</pose>
        </inertial>
        <collision name='front_right_wheel_link_collision'>
          <pose>0 0 0 1.5707963267948966 -0 0</pose>
          <geometry>
            <cylinder>
              <length>0.1143</length>
              <radius>0.1651</radius>
            </cylinder>
          </geometry>
          <surface>
            <contact>
              <ode>
                <kp>10000000</kp>
                <kd>1</kd>
              </ode>
            </contact>
            <friction>
              <ode>
                <mu>0.5</mu>
                <mu2>1</mu2>
                <slip1>0.00062</slip1>
                <slip2>0</slip2>
                <fdir1>0 0 1</fdir1>
              </ode>
            </friction>
            <bounce/>
          </surface>
        </collision>
        <visual name='front_right_wheel_link_visual'>
          <geometry>
            <mesh>
              <scale>1 1 1</scale>
              <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/MARBLE_HUSKY_SENSOR_CONFIG_4/tip/files/meshes/wheel.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <gravity>true</gravity>
        <velocity_decay/>
        <enable_wind>false</enable_wind>
      </link>
      <joint name='front_right_wheel_joint' type='revolute'>
        <child>front_right_wheel_link</child>
        <parent>base_link</parent>
        <axis>
          <xyz>0 1 0</xyz>
          <limit>
            <lower>-10000000000000000</lower>
            <upper>10000000000000000</upper>
            <effort>inf</effort>
            <velocity>inf</velocity>
            <stiffness>100000000</stiffness>
            <dissipation>1</dissipation>
          </limit>
          <dynamics>
            <spring_reference>0</spring_reference>
            <spring_stiffness>0</spring_stiffness>
            <damping>0</damping>
            <friction>0</friction>
          </dynamics>
        </axis>
        <pose>-0 -0 0 0 -0 0</pose>
      </joint>
      <link name='rear_left_wheel_link'>
        <pose>-0.256 0.2854 0.03282 -3.14159 1.4118 -3.14159</pose>
        <inertial>
          <mass>2.637</mass>
          <inertia>
            <ixx>0.024670000000000001</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.044110000000000003</iyy>
            <iyz>0</iyz>
            <izz>0.024670000000000001</izz>
          </inertia>
          <pose>0 0 0 0 -0 0</pose>
        </inertial>
        <collision name='rear_left_wheel_link_collision'>
          <pose>0 0 0 1.5707963267948966 -0 0</pose>
          <geometry>
            <cylinder>
              <length>0.1143</length>
              <radius>0.1651</radius>
            </cylinder>
          </geometry>
          <surface>
            <contact>
              <ode>
                <kp>10000000</kp>
                <kd>1</kd>
              </ode>
            </contact>
            <friction>
              <ode>
                <mu>0.5</mu>
                <mu2>1</mu2>
                <slip1>0.00062</slip1>
                <slip2>0</slip2>
                <fdir1>0 0 1</fdir1>
              </ode>
            </friction>
            <bounce/>
          </surface>
        </collision>
        <visual name='rear_left_wheel_link_visual'>
          <geometry>
            <mesh>
              <scale>1 1 1</scale>
              <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/MARBLE_HUSKY_SENSOR_CONFIG_4/tip/files/meshes/wheel.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <gravity>true</gravity>
        <velocity_decay/>
        <enable_wind>false</enable_wind>
      </link>
      <joint name='rear_left_wheel_joint' type='revolute'>
        <child>rear_left_wheel_link</child>
        <parent>base_link</parent>
        <axis>
          <xyz>-0 1 -0</xyz>
          <limit>
            <lower>-10000000000000000</lower>
            <upper>10000000000000000</upper>
            <effort>inf</effort>
            <velocity>inf</velocity>
            <stiffness>100000000</stiffness>
            <dissipation>1</dissipation>
          </limit>
          <dynamics>
            <spring_reference>0</spring_reference>
            <spring_stiffness>0</spring_stiffness>
            <damping>0</damping>
            <friction>0</friction>
          </dynamics>
        </axis>
        <pose>0 0 -0 0 -0 0</pose>
      </joint>
      <link name='rear_right_wheel_link'>
        <pose>-0.256 -0.2854 0.03282 3.14159 0.673419 3.14159</pose>
        <inertial>
          <mass>2.637</mass>
          <inertia>
            <ixx>0.024670000000000001</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.044110000000000003</iyy>
            <iyz>0</iyz>
            <izz>0.024670000000000001</izz>
          </inertia>
          <pose>0 0 0 0 -0 0</pose>
        </inertial>
        <collision name='rear_right_wheel_link_collision'>
          <pose>0 0 0 1.5707963267948966 -0 0</pose>
          <geometry>
            <cylinder>
              <length>0.1143</length>
              <radius>0.1651</radius>
            </cylinder>
          </geometry>
          <surface>
            <contact>
              <ode>
                <kp>10000000</kp>
                <kd>1</kd>
              </ode>
            </contact>
            <friction>
              <ode>
                <mu>0.5</mu>
                <mu2>1</mu2>
                <slip1>0.00062</slip1>
                <slip2>0</slip2>
                <fdir1>0 0 1</fdir1>
              </ode>
            </friction>
            <bounce/>
          </surface>
        </collision>
        <visual name='rear_right_wheel_link_visual'>
          <geometry>
            <mesh>
              <scale>1 1 1</scale>
              <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/MARBLE_HUSKY_SENSOR_CONFIG_4/tip/files/meshes/wheel.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <gravity>true</gravity>
        <velocity_decay/>
        <enable_wind>false</enable_wind>
      </link>
      <joint name='rear_right_wheel_joint' type='revolute'>
        <child>rear_right_wheel_link</child>
        <parent>base_link</parent>
        <axis>
          <xyz>-0 1 0</xyz>
          <limit>
            <lower>-10000000000000000</lower>
            <upper>10000000000000000</upper>
            <effort>inf</effort>
            <velocity>inf</velocity>
            <stiffness>100000000</stiffness>
            <dissipation>1</dissipation>
          </limit>
          <dynamics>
            <spring_reference>0</spring_reference>
            <spring_stiffness>0</spring_stiffness>
            <damping>0</damping>
            <friction>0</friction>
          </dynamics>
        </axis>
        <pose>0 0 -0 0 -0 0</pose>
      </joint>
      <pose>1.54075 0.86249 0.13228 -0 0 -2.39705</pose>
      <static>false</static>
      <self_collide>false</self_collide>
    </model>
    <model name='ground_plane'>
      <static>true</static>
      <link name='link'>
        <collision name='collision'>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>100</mu>
                <mu2>50</mu2>
              </ode>
            </friction>
            <bounce/>
            <contact/>
          </surface>
        </collision>
        <visual name='visual'>
          <cast_shadows>false</cast_shadows>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <material>
            <ambient>0.3 0.3 0.3 1</ambient>
            <diffuse>0.7 0.7 0.7 1</diffuse>
            <specular>0.01 0.01 0.01 1</specular>
          </material>
        </visual>
        <pose>0 0 0 0 -0 0</pose>
        <inertial>
          <pose>0 0 0 0 -0 0</pose>
          <mass>1</mass>
          <inertia>
            <ixx>1</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1</iyy>
            <iyz>0</iyz>
            <izz>1</izz>
          </inertia>
        </inertial>
        <enable_wind>false</enable_wind>
      </link>
      <pose>1.7587 0 0 0 -0 0</pose>
      <self_collide>false</self_collide>
    </model>
    <model name='stereo_camera'>
    <!-- Visualização do Conjunto Stereo -->
    <link name="stereo_visual">
      <!-- Representação visual: uma barra conectando as duas câmeras -->
      <pose>0 0 0 0 0 0</pose>
      <visual name="stereo_bar">
        <geometry>
          <box>
            <!-- Largura é a baseline (0.1 m), altura e profundidade são ajustáveis -->
            <size>0.1 0.2 0.05</size>
          </box>
        </geometry>
        <!-- Cor da barra -->
        <material>
          <ambient>0.3 0.3 0.3 1</ambient>
          <diffuse>0.6 0.6 0.6 1</diffuse>
        </material>
      </visual>
    </link>
    <static>true</static><!-- Câmera Esquerda -->
    <link name="left_camera_link">
      <pose>0 -0.1 0 0 0 0</pose> <!-- Posição relativa ao par stereo -->      
      <sensor name="left_camera" type="camera">
        <pose>0 0 0 0 0 0</pose>
        <camera>
          <horizontal_fov>1.047</horizontal_fov> <!-- Campo de visão horizontal -->
          <image>
            <width>1920</width>
            <height>1280</height>
            <format>R8G8B8</format>
          </image>
          <clip>
            <near>0.1</near>
            <far>100</far>
          </clip>
          <noise>
            <type>gaussian</type>
            <mean>0.0</mean>
            <stddev>0.007</stddev>
          </noise>
          <update_rate>30</update_rate>
        </camera>
        <always_on>true</always_on>
        <update_rate>30</update_rate>
        <visualize>true</visualize>
      </sensor>
    </link>
    <!-- Câmera Direita -->
    <link name="right_camera_link">
      <pose>0 0.1 0 0 0 0</pose> <!-- Posição relativa ao par stereo -->
      <sensor name="right_camera" type="camera">
        <pose>0 0 0 0 0 0</pose>
        <camera>
          <horizontal_fov>1.047</horizontal_fov> <!-- Campo de visão horizontal -->
          <image>
            <width>1920</width>
            <height>1280</height>
            <format>R8G8B8</format>
          </image>
          <clip>
            <near>0.1</near>
            <far>100</far>
          </clip>
          <noise>
            <type>gaussian</type>
            <mean>0.0</mean>
            <stddev>0.007</stddev>
          </noise>
          <update_rate>30</update_rate>
        </camera>
        <always_on>true</always_on>
        <update_rate>30</update_rate>
        <visualize>true</visualize>
      </sensor>
    </link>
    <pose>9.9253 -7.4372 2.907 0 0.328 2.5</pose>
    <static>true</static>
    <self_collide>false</self_collide>
  </model>
    <include>
      <uri>file:///home/gdaniel/ros2_ws/src/guided_navigation/models/room</uri>
      <name>urban_room_2way</name>
      <pose>0 0 0 0 -0 0</pose>
    </include>
    <light name='sun' type='directional'>
      <pose>0 0 10 0 -0 0</pose>
      <cast_shadows>true</cast_shadows>
      <intensity>1</intensity>
      <direction>-0.5 0.1 -0.9</direction>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <linear>0.01</linear>
        <constant>0.90000000000000002</constant>
        <quadratic>0.001</quadratic>
      </attenuation>
      <spot>
        <inner_angle>0</inner_angle>
        <outer_angle>0</outer_angle>
        <falloff>0</falloff>
      </spot>
    </light>
  </world>
</sdf>
